assets/js/src/core/modules/data-object/actions/add-object/use-add-object.tsx

```diff
@@ -0,0 +1,149 @@
+}
+
+export const useAddObject = (): UseAddObjectHookReturn => {
+  const { t } = useTranslation()
+  const classDefinitions = useClassDefinitions()
+  const modal = useFormModal()
+  const [addDataObjectMutation] = useDataObjectAddMutation()
+  const dispatch = useAppDispatch()
+
+  const getClassEntries = (node: TreeNodeProps): ItemType[] => {
+    let classHirachy: ItemType[] = []
+
+    const structuredClassDefinitions = [...classDefinitions]
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .reduce<Record<string, ClassDefinitionListItem[]>>((acc, classDefinition) => {
+      const groupName = _.isEmpty(classDefinition.group)
+        ? 'undefined'
+        : classDefinition.group
+
+      if (acc[groupName] === undefined) {
+        acc[groupName] = []
+      }
+
+      acc[groupName].push(classDefinition)
+
+      return acc
+    }, {})
+
+    if (structuredClassDefinitions.undefined !== undefined) {
+      classHirachy = structuredClassDefinitions.undefined.map(classDefinition => getDataObjectEntry(classDefinition, node))
+    }
+
+    for (const [group, classDefinitions] of Object.entries(structuredClassDefinitions)) {
+      if (group !== 'undefined') {
+        classHirachy.push({
+          label: group,
+          key: 'add-object-group-' + group,
+          icon: <Icon value={ 'folder' } />,
+          children: classDefinitions.map(classDefinition => getDataObjectEntry(classDefinition, node))
+        })
+      }
+    }
+
+    return classHirachy
+  }
+
+  const getDataObjectEntry = (classDefinition: ClassDefinitionListItem, node: TreeNodeProps): ItemType => {
+    return {
+      label: classDefinition.name,
+      key: classDefinition.id,
+      icon: <Icon { ...classDefinition.icon } />,
+      onClick: () => {
+        const id = parseInt(node.id)
+        const parentId = parseInt(node.id)
+        createDataObject(id, classDefinition, parentId)
+      }
+    }
+  }
+
+  const createDataObject = (
+    id: number,
+    classDefinition: ClassDefinitionListItem,
+    parentId: number,
+    onFinish?: (newName: string) => void
+  ): void => {
+    modal.input({
+      title: t('data-object.create-data-object', { className: classDefinition.name }),
+      label: t('form.label.new-item'),
+      rule: {
+        required: true,
+        message: t('form.validation.required')
+      },
+      onOk: async (value: string) => {
+        await createDataObjectMutation(classDefinition.id, value, parentId)
+        onFinish?.(value)
+      }
+    })
+  }
+
+  const createDataObjectMutation = async (
+    classId: string,
+    name: string,
+    parentId: number
+  ): Promise<void> => {
+    const createDataObjectTask = addDataObjectMutation({
+      parentId,
+      dataObjectAddParameters: {
+        key: name,
+        classId,
+        type: 'object'
+      }
+    })
+
+    try {
+      const response = await createDataObjectTask
+
+      if (response.error !== undefined) {
+        trackError(new ApiError(response.error))
```

Comment:
Please add a translation key after this issue is solved by the backend team:

https://github.com/pimcore/studio-backend-bundle/issues/903